### ISSUE TITLE
Float and double tensors randomly initialized with the same seed get different values for size >= 16

### ISSUE THREAD

**Issue by aperezlebel** [pytorch/pytorch#79703]
### 🐛 Describe the bug

Hello, when randomly initializing with the same seed two size-n tensors with dtypes float and double, they get different values for n >= 16. For n <= 15 they are equal as expected.

### Sample code to reproduce the problem
```python
import torch

n = 16
t1 = torch.zeros(n, dtype=torch.float)
t2 = torch.zeros(n, dtype=torch.double)

torch.manual_seed(0)
t1.normal_(0, 1)
torch.manual_seed(0)
t2.normal_(0, 1)

print(t1)
print(t2)
```

### Expected result
`t1` and `t2` should contain the same values.

### Obtained result
`t1` and `t2` have different values for n >= 16. For example with n=16:
```python
tensor([-1.1258, -1.1524, -0.2506, -0.4339,  0.8487,  0.6920, -0.3160, -2.1152,
         0.3223, -1.2633,  0.3500,  0.3081,  0.1198,  1.2377,  1.1168, -0.2473])
tensor([-2.3104, -0.3733, -1.0608,  0.9995, -0.8840, -1.2755, -0.6232, -0.8664,
        -1.2956,  1.5236,  0.3237,  2.0177,  1.1357, -1.2269,  0.0714,  0.3380],
       dtype=torch.float64)
```
`t1` and `t2` have equal values for n <= 15. For example with n=15:
```python
tensor([ 1.5410, -0.2934, -2.1788,  0.5684, -1.0845, -1.3986,  0.4033,  0.8380,
        -0.7193, -0.4033, -0.5966,  0.1820, -0.8567,  1.1006, -1.0712])
tensor([ 1.5410, -0.2934, -2.1788,  0.5684, -1.0845, -1.3986,  0.4033,  0.8380,
        -0.7193, -0.4033, -0.5966,  0.1820, -0.8567,  1.1006, -1.0712],
       dtype=torch.float64)
```

### Versions

```
Collecting environment information...
PyTorch version: 1.11.0.post2
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.3 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: 10.0.0-4ubuntu1 
CMake version: Could not collect
Libc version: glibc-2.31

Python version: 3.10.5 | packaged by conda-forge | (main, Jun 14 2022, 07:06:46) [GCC 10.3.0] (64-bit runtime)
Python platform: Linux-5.14.0-1042-oem-x86_64-with-glibc2.31
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.22.4
[pip3] torch==1.11.0.post2
[conda] libblas                   3.9.0            15_linux64_mkl    conda-forge
[conda] libcblas                  3.9.0            15_linux64_mkl    conda-forge
[conda] liblapack                 3.9.0            15_linux64_mkl    conda-forge
[conda] mkl                       2022.1.0           h84fe81f_915    conda-forge
[conda] numpy                     1.22.4          py310h4ef5377_0    conda-forge
[conda] pytorch                   1.11.0          cpu_py310h75c9ab6_2    conda-forge
```
